osi_logicallane.proto

```diff
@@ -0,0 +1,634 @@
+    // Example:
+    //                |---------\
+    //                |          \
+    //  /---boundary--|           \------------
+    // /
+    // ---------- reference line --------------
+    // At the place where the boundary suddenly increases, two points will have
+    // the same S coordinate.
+    //
+    // If the boundary approximates a curve (e.g. a cubic function in
+    // OpenDRIVE), the points must be chosen in a way that the lateral distance
+    // to the ideal line does not exceed 5cm. As shown in the following image:
+    //
+    // \image html line_approximation_error.svg "Approximation error"
+    // Approximation error green line.
+    //
+    // The Z error (difference in Z height between boundary_line and the "real"
+    // line) must also not exceed 5cm.
+    //
+    // Note: if two lanes have different Z heights (e.g. a driving lane is
+    // beside a sidewalk, where the sidewalk is 10cm higher than the road),
+    // then these lanes cannot share a boundary, since their boundaries have
+    // different Z heights.
+    //
+    repeated LogicalBoundaryPoint boundary_line = 2;
+
+    // The reference line for this boundary.
+    //
+    // The reference line is used as a coordinate system for this boundary.
+    // All points of this LogicalLaneBoundary must have S coordinates in the
+    // range [sStart,sEnd].
+    //
+    // The reference line should roughly have the same shape as the boundary (so
+    // roughly parallel to the lane middle), so that S coordinates continually
+    // increase along the boundary.
+    //
+    // \rules
+    // refers_to: ReferenceLine
+    // \endrules
+    //
+    optional Identifier reference_line_id = 3;
+
+    // Reference to the physical lane boundary or boundaries that make up this
+    // logical boundary.
+    //
+    // Rules and notes:
+    // - This list is empty if there are no physical lane boundaries to delimit
+    //   a lane.
+    // - In the common case, this will contain one physical boundary.
+    // - This list contains several lane boundaries if there are several physical
+    //   lane boundaries at one S position (e.g. both a broken and a solid
+    //   line).
+    // - If there are several lane boundaries, they must be listed in increasing
+    //   T order (i.e. from right to left in reference line direction).
+    //   Rationale: this makes it easier to determine e.g. rules on lane
+    //   changes, which depend on the T order of the lanes.
+    // - Whenever physical lane boundaries begin or end, or switch their T
+    //   position (if there are multiple physical lane boundaries), a new
+    //   LogicalLaneBoundary must be created.
+    // - The referenced LaneBoundary objects may be longer than the
+    //   LogicalLaneBoundary which references them, but must never be shorter.
+    //
+    // Example:
+    //      Lane 1
+    // --------a------------------ - - - -c- - - -
+    // - - - -b- - - -
+    //      Lane 2
```

Comment:
As discussed, a bit more explanation e.g. where goes the reference line in this example...